plugin/src/main/java/com/google/errorprone/xplat/checker/JodaTimeLocal.java

```diff
@@ -0,0 +1,189 @@
+                  .map(
+                      typeName ->
+                          Matchers.constructor()
+                              .forClass(typeName)
+                              .withParameters("org.joda.time.DateTimeZone"))
+                  .collect(toImmutableList())),
+          Matchers.constructor()
+              .forClass("org.joda.time.LocalDateTime")
+              .withParameters("long", "org.joda.time.DateTimeZone"),
+          Matchers.constructor()
+              .forClass("org.joda.time.LocalDateTime")
+              .withParameters("java.lang.Object", "org.joda.time.DateTimeZone"));
+
+  private Matcher<ExpressionTree> methodMatcher(String key) {
+    return Matchers.anyOf(
+        CLASS_NAMES.get(key).stream()
+            .map(
+                methodName ->
+                    Matchers.instanceMethod().onExactClass(key).named(methodName))
+            .collect(toImmutableList()));
+  }
+
+  private Description.Builder message(Tree tree, String arg) {
+    return buildDescription(tree)
+        .setMessage(
+            String.format("The use of %s is banned from cross platform development due to"
+                + " incompatibilities.", arg));
+  }
+
+  private Description messageFix(Tree tree, String arg, int start, int end,
+      String replacement) {
+    return message(tree, arg)
+        .addFix(
+            SuggestedFix.builder()
+                .addImport("org.joda.time.DateTimeZone")
+                .replace(
+                    start,
+                    end,
+                    replacement)
+                .build())
+        .build();
+  }
+
+
+  @Override
+  public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+    ExpressionTree recv = ASTHelpers.getReceiver(tree);
+    String recvSrc = state.getSourceForNode(recv);
+    Symbol symbol = ASTHelpers.getSymbol(tree);
+    List<? extends ExpressionTree> arguments = tree.getArguments();
+
+    String argument;
+    if (arguments.isEmpty()) {
+      argument = "DateTimeZone.getDefault()";
+    } else {
+      argument = state.getSourceForNode(arguments.get(0));
+    }
+
+    if (symbol != null && recvSrc != null && recv != null) {
+      if (methodMatcher("org.joda.time.LocalDateTime").matches(tree, state)) {
+
+        return messageFix(tree, symbol.toString(), ((JCTree) recv).getStartPosition(),
+            state.getEndPosition(tree),
+            String.format("new DateTime(%s.getYear(), %<s.getMonthOfYear(),"
+                    + " %<s.getDayOfYear(), %<s.getHourOfDay(),"
+                    + " %<s.getMinuteOfHour(), %<s.getSecondOfMinute(),"
+                    + " %<s.getMillisOfSecond(), %s)",
+                recvSrc, argument));
+
+      } else if (methodMatcher("org.joda.time.LocalTime").matches(tree, state)) {
+
+        return messageFix(tree, symbol.toString(), ((JCTree) recv).getStartPosition(),
+            state.getEndPosition(tree), String.format("new DateTime().now(%s).withTime(%s)",
```

Comment:
If we only target Java, since `now(DateTimeZone)` is a static method, this could just be `DateTime.now(...)`. This being said, not all platforms support that. It turns out that only `new DateTime().toDateTime(DateTimeZone)` is supported everywhere. So we'll have to replace the `now` with `toDateTime()` everywhere. My apologies for not mentioning this earlier…

Reply:
Makes sense. I have updated the fix with this change.